agent/lib_aws_sdk_php.c

```diff
@@ -150,24 +149,15 @@ void nr_lib_aws_sdk_php_sqs_handle(nr_segment_t* segment,
   nr_segment_message_end(&segment, &message_params);
 
   nr_free(command_arg_value);
-
-  /*
-   * These are the message_params params that were strduped in
-   * nr_lib_aws_sdk_php_sqs_parse_queueurl
-   */
-  nr_free(cloud_attrs.cloud_region);
-  nr_free(message_params.destination_name);
-  nr_free(cloud_attrs.cloud_account_id);
 }
 
```

Comment:
I think `nr_lib_aws_sdk_php_sqs_parse_queueurl` should set the fields in `message_params` and `cloud_attrs` to `NULL` at the start as the function can exit w/o actually storing these values.  This could lead to issues when the caller uses these uninitialized values.

Reply:
The values are already being initialized [here](https://github.com/newrelic/newrelic-php-agent/pull/1003/files#diff-b5410abe33a951704c8600c7b0b214d3c431cee94dcb3089567ec0ec9f328fc1R94-R99).
All cloud_attrs start as NULL, all message_params not explicitly set are set to NULL during that initialization.

Reply:
I guess it is a question of expectations - my initial expectation was that this function to `NULL` those fields out if the `queryurl` was unparsable to reflect no data was found to fill them in properly.

Reply:
Those 3 fields are already set to NULL by the caller who NULLs all the unknown fields prior to finding values for them.  Caller is responsible for initializing the structs, not parseQ.
parseQ should only modify the fields if it has something meaninful to add; otherwise, it should leave them untouched.  At some point in future, we might give any of these values "unknown" or something as default then pass it to parseQ, we wouldn't want parseQ to ovewrite the default with NULL.  parseQ is only reponsible setting the values if they are known; otherwise, caller is responsible for setting struct values.